tests/PHPStan/Analyser/data/integer-range-types.php

```diff
@@ -281,6 +281,16 @@ public function math($i, $j, $z, $pi, $r1, $r2, $r3, $rMin, $rMax, $x, $y) {
 
 		assertType('5|10|15|20|30', $x / $y);
 
+		assertType('float|int<0, max>', $rMax / $rMax);
```

Comment:
https://www.wolframalpha.com/input/?i=%28interval%5B5%2C%E2%88%9E%5D++%2F+interval%5B5%2C%E2%88%9E%5D%29

Reply:
In Psalm I chose not to go this route and just inferred `int|float`. There's a point where the line is too blurry. In this case, whatever operations or expectation you have that would justify knowing the range will get invalidated by `float`. You can't deduce the result is an int, nor a float, you can't deduce it's positive.

Reply:
@ondrejmirtes I'm not sure I understood what you meant by that

Reply:
to sum up: the assertion for this line should be changed to `float|int` ?

Reply:
@orklah It's harder for this to make `float|int` instead of the current result. I'd leave it the same.

Reply:
@orklah I mentioned the cast-cases in https://github.com/phpstan/phpstan/issues/5731#issuecomment-937520060

not sure how hard it would be to get the remaing failing case work though